Bluetooth controller: gcc warning: ‘ad_len_chain’ may be used uninitialized

**Describe the bug**
gcc 12 complains about a possible non-initialized variable use in the BT controller


**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf5340bsim_nrf5340_cpunet ../samples/bluetooth/hci_rpmsg/ -DCONF_FILE="nrf5340_cpunet_bis-bt_ll_sw_split.conf"
3. ninja
4. See error

**Expected behavior**
No build warnings

**Impact**
* Twister failures with gcc 12
* An actual possible bug
* Compiler warnings when building manually

**Logs and console output**
```
[112/227] Building C object zephyr/subsys/bluetooth/controller/CMakeFiles/subsys__bluetooth__controller.dir/ll_sw/ull_adv_sync.c.obj
zephyr/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c: In function ‘ll_adv_sync_ad_data_set’:
zephyr/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:642:48: warning: ‘ad_len_chain’ may be used uninitialized [-Wmaybe-uninitialized]
  642 |                 if ((ter_len + ad_len_overflow + len) >
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
zephyr/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:257:17: note: ‘ad_len_chain’ was declared here
  257 |         uint8_t ad_len_chain;
      |                 ^~~~~~~~~~~~
```

**Environment:**
 - OS: Ubuntu 22.04
 - Toolchain: gcc 12.3.0
 - Commit SHA or Version used: Main as of now 3f1eed17f83a93e6dcd731d2a3b77b48f985d68c

--- a/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c
+++ b/subsys/bluetooth/controller/ll_sw/ull_adv_sync.c
@@ -403,6 +403,14 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 
 				/* No AD data overflow */
 				ad_len_overflow = 0U;
+
+				/* No chain PDU.
+				 * Note: Not required to assign as referencing
+				 * is guarded by the fact that ad_len_overflow
+				 * is zero; having the below to make compilers
+				 * not complain of uninitialized variable.
+				 */
+				ad_len_chain = 0U;
 			}
 		}
 	} else {
@@ -484,7 +492,16 @@ uint8_t ll_adv_sync_ad_data_set(uint8_t handle, uint8_t op, uint8_t len,
 			/* Proceed to add chain PDU */
 			err = 0U;
 		} else {
+			/* No AD data overflow */
 			ad_len_overflow = 0U;
+
+			/* No chain PDU.
+			 * Note: Not required to assign as referencing is
+			 * guarded by the fact that ad_len_overflow is zero;
+			 * having the below to make compilers not complain of
+			 * uninitialized variable.
+			 */
+			ad_len_chain = 0U;
 		}
 #endif /* CONFIG_BT_CTLR_ADV_SYNC_PDU_LINK */
 	}

